schema/integer.go

```diff
@@ -12,8 +12,8 @@ type Integer struct {
 	Boundaries *Boundaries
 }
 
-// Validate validates and normalize integer based value.
-func (v Integer) Validate(value interface{}) (interface{}, error) {
+// Prepare TODO
+func (v Integer) Prepare(value interface{}) (interface{}, error) {
 	if f, ok := value.(float64); ok {
 		// JSON unmarshaling treat all numbers as float64, try to convert it to
 		// int if not fraction.
```

Comment:
`I do think that in the `Prepare` phase, we should allow `float` values to be converted to `integer`. 
I.e. `?filter={someint: {$gt: 23.44}}` should be valid. However accepting `float` in `PUT,POST,PATCH` should be forbidden.